submissions/ibrahimHamouda/script.js

```diff
@@ -0,0 +1,64 @@
+/* first */
+const editor = {
+  title: 'my first blog',
+  getUpperTitle() {
+    return this.title.toUpperCase();
+  }
+};
+
+const getTitle = editor.getUpperTitle.bind(editor);
+console.log(getTitle())
+// Write code that ensures getTitle() works as expected and logs "MY FIRST BLOG"
+
+/* ------------------------------------ */
+
+/* 2 1-use arrow function */
+const formHandler = {
+  value: 'initial',
+  onChange:(newValue) =>{
+	 formHandler.value = newValue;
+  }
+}
+function simulateInputChange(callback) {
+  callback('updated');
+}
+simulateInputChange(formHandler.onChange)
+
+
+/* 1-Predict what formHandler.value will be */
+console.log(formHandler.value)
+
+/* 2 1-use bind  */
+const formHandler1 = {
+  value: 'initial',
+  onChange(newValue) {
+    this.value = newValue;
+  }
+};
+function simulateInputChange(callback) {
+  callback('updated');
+}
+simulateInputChange(formHandler1.onChange.bind(formHandler1));
+console.log(formHandler1.value)
+
```

Comment:
It's nice from u to solve it in two different ways, great!

Reply:
and please predict what is the value in line 42+ explain why u use bind or the arrow function in both solutions